### PR TITLE
[GEOS-10779] Upgrade to Spring 5.3.23 and Spring Security 5.7.3

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOS-10779](https://osgeo-org.atlassian.net/browse/GEOS-10779)

stream changes

jsontype

more test character set

mock charset issue figured out

pr cleanup

wms test fixes

rest tests part 1

only grabbing mimetype

gwc test fixes

most of rest config

cleanup

app schema mimetype

changed to non deprecated filter

another deprected call

fixed issues with replaced deprecated filter

another security context deprecation fix

cleanup

rename mime helper and monitor extension stuff

fixed deprecated and misconfigured accepts

multidimensional content types

more deprecated removal

stringutil deprecated

more stringutil

cleanup

fixed geofence tests

fixed spotbugs null issue

more deprecation

cleanup

community cleanup from deprecations

pr cleanup

securitycontextfilter note added

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Review by mprins — COMMENTED** [review]
Some small remarks/fixes inline, untested, but LGTM.

File: src/pom.xml [hunk 1]
```diff
@@ -92,9 +92,9 @@
     <gt.version>29-SNAPSHOT</gt.version>
     <gwc.version>1.23-SNAPSHOT</gwc.version>
     <jts.version>1.19.0</jts.version>
-    <spring.version>5.2.22.RELEASE</spring.version>
-    <spring.security.version>5.1.13.RELEASE</spring.security.version>
-    <servlet-api.version>3.0.1</servlet-api.version>
+    <spring.version>5.3.23</spring.version>
```
[the remark this answers is not in the document]
Comment (jodygarnett) [reply src/pom.xml:95]: That slightly more than a suggestion; it is changing scope of the PR :)

[the remark this answers is not in the document]
Comment (aaime) [reply src/pom.xml:95]: It means we have to re-rest everything (manually, for community modules) and handle new deprecations

**Comment by jodygarnett** [comment]
Testing locally with @mprins proposed spring 5.7.3 and spring security 5.7.6 ... and it built cleanly :)

File: src/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mprins) [reply src/pom.xml:95]: Knowing Spring's extremely conservative approach that is very unlikely.

for reference:
- https://github.com/spring-projects/spring-framework/releases/tag/v5.3.24
- https://github.com/spring-projects/spring-framework/releases/tag/v5.3.25

Both fix a number of bugs (that may or may not affect Geoserver) that are present in 5.2.23

[the remark this answers is not in the document]
Comment (turingtestfail) [reply src/pom.xml]: Rebased, retested, and updated.

File: src/pom.xml [hunk 2]
```diff
@@ -92,9 +92,9 @@
...
+    <spring.security.version>5.7.3</spring.security.version>
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply src/pom.xml]: Rebased, retested, and updated.

**Review by mprins — APPROVED** [review]

**Pull request merged**